openpaas-james/mailbox/opendistro/src/test/java/org/apache/james/mailbox/opendistro/DockerOpenDistro.java

```diff
@@ -0,0 +1,361 @@
+            GenericContainer<?> container = new GenericContainer<>(imageName);
+            container.withTmpFs(ImmutableMap.of("/usr/share/elasticsearch/data", "rw,size=200m"));
+            container.withExposedPorts(ES_HTTP_PORT);
+            container.withEnv("discovery.type", "single-node");
+            container.withCopyFileToContainer(MountableFile.forClasspathResource("elasticsearch.yml"), "/usr/share/elasticsearch/config/elasticsearch.yml");
+
+            return new DockerContainer(container)
+                .withAffinityToContainer()
+                .waitingFor(new HostPortWaitStrategy().withRateLimiter(RateLimiters.TWENTIES_PER_SECOND));
+        }
+
+        private final DockerContainer eSContainer;
+
+        public NoAuth() {
+            this(OPENDISTRO_1_13_1);
+        }
+
+        public NoAuth(String imageName) {
+            this.eSContainer = defaultContainer(imageName);
+        }
+
+        public NoAuth(DockerContainer eSContainer) {
+            this.eSContainer = eSContainer;
+        }
+
+        public void start() {
+            if (!isRunning()) {
+                eSContainer.start();
+            }
+        }
+
+        public void stop() {
+            eSContainer.stop();
+        }
+
+        public int getHttpPort() {
+            return eSContainer.getMappedPort(ES_HTTP_PORT);
+        }
+
+        public String getIp() {
+            return eSContainer.getHostIp();
+        }
+
+        public Host getHttpHost() {
+            return Host.from(getIp(), getHttpPort());
+        }
+
+        public void pause() {
+            eSContainer.pause();
+        }
+
+        public void unpause() {
+            eSContainer.unpause();
+        }
+
+        @Override
+        public boolean isRunning() {
+            return eSContainer.isRunning();
+        }
+    }
+
+    class WithAuth implements DockerOpenDistro {
+
+        private static final org.slf4j.Logger LOGGER = LoggerFactory.getLogger(WithAuth.class);
+
+        private static final String DEFAULT_USERNAME = "elasticsearch";
+        private static final String DEFAULT_PASSWORD = "secret";
+        public static final Credential DEFAULT_CREDENTIAL =
+            Credential.of(DEFAULT_USERNAME, DEFAULT_PASSWORD);
+
+        private final DockerOpenDistro.NoAuth elasticSearch;
+        private final DockerContainer nginx;
+        private final Network network;
+
+        public WithAuth() {
+            this(OPENDISTRO_1_13_1);
+        }
+
+        WithAuth(String imageName) {
+            this.network = Network.newNetwork();
+            this.elasticSearch = new DockerOpenDistro.NoAuth(
+                DockerOpenDistro.NoAuth
+                    .defaultContainer(imageName)
+                    .withLogConsumer(frame -> LOGGER.debug("[ElasticSearch] " + frame.getUtf8String()))
+                    .withNetwork(network)
+                    .withNetworkAliases("elasticsearch"));
+
+            this.nginx = new DockerContainer(
```

Comment:
As said before, let's remove this nginx container. This was for simulating TLS + auth connections as ES auth plugin are proprietary. However with Opendistro those are available, thus making the use of nginx obsolete here.

Might need to rework the logic around it likely though